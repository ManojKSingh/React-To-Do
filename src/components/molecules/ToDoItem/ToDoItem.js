import React from 'react';
import styled from 'styled-components';
import styles from './ToDoItem.style';
import Button from '../../atoms/Button';

const ToDoItem = ({children, isDone, itemId, onRemove, markDone, className, ...otherProps}) => {
    return (
        <div className={className}>
            <div className={`item-text ${isDone ? 'marked': ''}`} onClick={() => markDone(itemId)}>{children}</div>
            <Button styleType='remove' onClick={() => onRemove(itemId)}>x</Button>
        </div>
    );
}

export default styled(ToDoItem)`
    ${styles}
`;