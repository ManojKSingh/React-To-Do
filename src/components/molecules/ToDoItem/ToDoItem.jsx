import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import styles from './ToDoItem.style';
import { Button } from '../../atoms/Button';

const ToDoItem = ({itemId, isDone, onRemove, markDone, className, children, ...otherProps}) => {
    return (
        <div className={`${className} ${isDone ? 'disable' : ''}`} onClick={() => markDone(itemId)}>
            <div className={`item-text ${isDone ? 'marked': ''}`}>{children}</div>
            <Button styleType='remove' onClick={(e) => onRemove(e, itemId)}>x</Button>
        </div>
    );
}

ToDoItem.defaultProps = {
  className: ''
};

ToDoItem.propTypes = {
	itemId: PropTypes.string,
  isDone: PropTypes.bool,
  onRemove: PropTypes.func,
  markDone: PropTypes.func
};

const container = styled(ToDoItem)`
    ${styles}
`;

export {container as ToDoItem}