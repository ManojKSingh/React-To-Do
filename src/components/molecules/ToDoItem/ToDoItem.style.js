import { css } from 'styled-components';

const AppStyles = css`
    display: flex;
    align-items: center;
    padding: 15px 5px 15px 15px;
    margin: 10px 0;
    border: 1px solid ${props => props.theme.itemBorderColor};
    &.disable {
        background-color: ${props => props.theme.disableBgColor};
        color: ${props => props.theme.disableTextColor};
    }
    .item-text {
        text-align: justify;
        padding-right: 10px;
        cursor: pointer;
    }
    .item-text.marked {
        text-decoration: line-through;
    }
`;

export default AppStyles;