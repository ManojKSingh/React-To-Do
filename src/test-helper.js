import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import themes from '../src/themes';

const theme = themes['light'];

export function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}



