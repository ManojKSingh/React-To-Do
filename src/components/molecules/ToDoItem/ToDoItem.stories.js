import React from 'react';
import ToDoItem from './ToDoItem';
import { action } from '@storybook/addon-actions';

export default {
  title: 'Molecule/ToDoItem',
  component: ToDoItem,
};

export const toDoItem = () => (<ToDoItem isDone={false}
                                itemId='1'
                                markDone={action('Action to mark done')}
                                onRemove={action('Action to remove item')}>
                                Create Story Book
                            </ToDoItem>);