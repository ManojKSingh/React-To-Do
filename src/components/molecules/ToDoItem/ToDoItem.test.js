import React from "react";
import { renderWithTheme } from "../../../test-helper";
import {ToDoItem} from "./ToDoItem";

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe("<ToDoItem/>", () => {
  it("renders correctly", () => {
    const mockOnRemove = jest.fn();
    const mockMarkDone = jest.fn();
    const tree = renderWithTheme(
      <ToDoItem
        itemId={1}
        isDone={false}
        onRemove={mockOnRemove}
        markDone={mockMarkDone}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when mark done", () => {
    const mockOnRemove = jest.fn();
    const mockMarkDone = jest.fn();
    const tree = renderWithTheme(
      <ToDoItem
        itemId={1}
        isDone={true}
        onRemove={mockOnRemove}
        markDone={mockMarkDone}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
