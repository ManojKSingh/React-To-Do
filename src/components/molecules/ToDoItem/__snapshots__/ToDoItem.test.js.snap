// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToDoItem/> renders correctly 1`] = `
<div
  className="sc-AxirZ bJALbz "
  onClick={[Function]}
>
  <div
    className="item-text "
  />
  <button
    className="sc-AxjAm eiAZdM"
    onClick={[Function]}
  >
    x
  </button>
</div>
`;

exports[`<ToDoItem/> renders correctly when mark done 1`] = `
<div
  className="sc-AxirZ bJALbz disable"
  onClick={[Function]}
>
  <div
    className="item-text marked"
  />
  <button
    className="sc-AxjAm eiAZdM"
    onClick={[Function]}
  >
    x
  </button>
</div>
`;
